pkg/rancher-components/src/components/Form/Checkbox/Checkbox.vue

```diff
@@ -114,6 +114,9 @@ export default Vue.extend({
   },
 
   computed: {
+    modelValue() {
```

Comment:
A data prop would be more appropriate for `modelValue` if we intend to mutate it via `v-model`.

We should consider computed props to be getters. We can provide a setter if we need it, but I don't see the benefit in this scenario

https://v2.vuejs.org/v2/guide/computed.html#Computed-Setter

Reply:
Explicitly specifying the getter still doesn't fix the issue at hand. We are still mutating `modelValue` without a setter as well as mutating the `value` prop. The computed prop implementation works, but just because we are mutating `value` in a roundabout way and the vue template compiler isn't picking up on it. 

I don't think we need to be tracking state locally at all. I hope you don't mind, but I took a closer look and ended up amending this PR so that we could address the root issue.